src/track/track.cpp

```diff
@@ -794,6 +805,39 @@ QList<CuePointer> Track::getCuePoints() const {
 void Track::setCuePoints(const QList<CuePointer>& cuePoints) {
     //qDebug() << "setCuePoints" << cuePoints.length();
     QMutexLocker lock(&m_qMutex);
+    setCuePointsMarkDirtyAndUnlock(
+            &lock,
+            cuePoints);
+}
+
+void Track::importCues(
+        const QList<mixxx::CueInfo>& cueInfos) {
+    QMutexLocker lock(&m_qMutex);
+    if (m_streamInfo) {
+        // Replace existing cue points with imported cue
+        // points immediately
+        importCuesMarkDirtyAndUnlock(
+                &lock,
+                cueInfos);
+    } else {
+        kLogger.debug()
+                << "Deferring import of"
+                << cueInfos.size()
+                << "cue(s) until actual sample rate becomes available";
+        m_importCuesPending.append(cueInfos);
+        // Clear all existing cue points, that are supposed
+        // to be replaced with the imported cue points soon.
+        setCuePointsMarkDirtyAndUnlock(
+                &lock,
+                QList<CuePointer>{});
+    }
+}
+
+void Track::setCuePointsMarkDirtyAndUnlock(
+        QMutexLocker* pLock,
+        const QList<CuePointer>& cuePoints) {
+    DEBUG_ASSERT(pLock);
+    DEBUG_ASSERT(m_importCuesPending.isEmpty());
```

Comment:
I'm hitting this debug assertion when I browse a directory with tracks that contain Serato Metadata. In line 827 you append to the `m_importCuesPending` list, then call this function and trigger the DEBUG_ASSERT.

Reply:
Thanks, fixed. The debug assertion was too restrictive. I also added a detailed comment why and what we are checking now.